api/src/services/authorization.js

```diff
@@ -312,22 +291,9 @@ const findContactsByReplicationKeys = (replicationKeys) => {
       });
       docIds = _.uniq(docIds);
 
-      return Promise.all([
-        db.medic.allDocs({ keys: docIds, include_docs: true }),
-        ...getAllTombstones(docIds),
-      ]);
+      return db.medic.allDocs({ keys: docIds, include_docs: true });
     })
-    .then(results => {
-      const contacts = results.reduce((acc, result) => {
-        if (!result || !result.rows || !result.rows.length) {
-          return acc;
-        }
-
-        acc.push(...result.rows.map(row => row.doc).filter(doc => doc));
-        return acc;
-      }, []);
-      return contacts;
-    });
+    .then(results => results.rows.map(row => row.doc).filter(doc => doc));
```

Comment:
As this is a hot part of the code I wonder if it's worth doing some micro-optimisations, for example, this line iterates over the results twice but it could be done in one loop but it would be less readable. This is similar in the first few lines of the `findContactsByReplicationKeys` function where we do a without, a uniq, and a map (ie: 3 iterations).

Not necessary for this iteration but it'd be cool to do a run over the hot code sometime and see if we can get some real benefits.

Reply:
I agree. This is already a pretty big PR so I'm reluctant to do refactors here. 

`findContactsByReplicationKeys` is expected to iterate over a small number of elements - because it's a bottom-up search - where we're only getting the relevant contacts from the total pool of contacts that the user can see.

Reply:
Fair enough. Let's leave optimisations till later.